src/components/App.tsx

```diff
@@ -528,6 +530,26 @@ function App() {
       return newEdges;
     });
 
+    // Generate auto-label for parentNode, if unset
+
+    if (parentNode.data.label === parentNode.data.fluxNodeType) {
+      const autoLabel =
+        parentNode.data.text.length > MAX_AUTOLABEL_LENGTH
+          ? parentNode.data.text
+              .slice(0, MAX_AUTOLABEL_LENGTH)
+              .split(" ")
+              .slice(0, -1)
+              .join(" ") + " ..."
+          : parentNode.data.text;
+
+      setNodes((nodes) =>
+        modifyFluxNodeLabel(nodes, {
+          id: parentNode.id,
+          label: autoLabel,
+        })
+      );
+    }
```

Comment:
this is efficient and works but can we do this in onType? would just feel more seamless i think

Reply:
mmmh I guess we could, let me look into it

Reply:
would also like to ensure we handle regenerating GPT nodes, and GPT nodes being edited manually

Reply:
> mmmh I guess we could, let me look into it

worst case this is still a major improvement over the status quo

Reply:
> would also like to ensure we handle regenerating GPT nodes

If you mean auto regenerating label for existing GPT nodes, that should be covered (unless generated text is less than 32 chars, in which case it gets treated as a custom label and doesn't auto-generate).

> and GPT nodes being edited manually

When we add the onType with the prompt? Yeah I think that if we add that we should make it work for both User and GPT nodes

Reply:
> If you mean auto regenerating label for existing GPT nodes, that should be covered (unless generated text is less than 32 chars, in which case it gets treated as a custom label and doesn't auto-generate).

oh yeah you're totally right my bad. 

> When we add the onType with the prompt? Yeah I think that if we add that we should make it work for both User and GPT nodes

yup yup — also system nodes ideally